tests/e2e/specs/schemas.spec.ts

```diff
@@ -1,238 +1,383 @@
+ * 4. Try to evolve the schema to a new version
+ *    a. Valid/compatible schema update
+ *    b. Invalid/incompatible schema update
+ * 5. Clean up by deleting the subject
  */
-async function uploadSchema(page: any, subjectName: string) {
-  await page.getByLabel(/Schemas.*Section/).hover();
-  await expect(page.getByLabel("Upload Schema to Schema Registry", { exact: true })).toBeVisible();
-  await page.getByLabel("Upload Schema to Schema Registry", { exact: true }).click();
-
-  await expect(page.getByPlaceholder("Select a file")).toBeVisible();
-  await page.getByPlaceholder("Select a file").click();
-  // Select the first option.
-  await page.keyboard.press("Enter");
-
-  await page.keyboard.type("AVRO");
-  await page.keyboard.press("Enter");
-
-  await page.keyboard.type(subjectName);
-  await page.keyboard.press("Enter");
-}
-
-async function deleteSubject(page: any, subjectName: string) {
-  await page.keyboard.press("Shift+ControlOrMeta+P");
-  await page.keyboard.type("Delete Subject");
-  await page.keyboard.press("Enter");
-
-  await expect(page.getByPlaceholder("Select existing subject")).toBeVisible();
-  await page.keyboard.type(subjectName);
-  await page.keyboard.press("Enter");
-
-  await page.getByLabel("Hard Delete").click();
-
-  const validationMessage = await page
-    .getByPlaceholder(/Enter "hard .*" to confirm/)
-    .getAttribute("placeholder");
-  const match = validationMessage?.match(/Enter "hard (.*)" to confirm/);
-  if (match) {
-    const requiredText = match[1];
-    await page.keyboard.type(`hard ${requiredText}`);
-    await page.keyboard.press("Enter");
-  }
 
-  await expect(
-    page.getByText(/Subject customer-.*-value and.*hard deleted./, { exact: true }).first(),
-  ).toBeVisible();
-}
+test.describe("Schema Management", () => {
+  let resourcesView: ResourcesView;
+  // this is set after the connections are set up based on their beforeEach hooks
+  let schemasView: SchemasView;
 
-test.describe("Schema related functionality", () => {
   let subjectName: string;
+  // most tests only create one schema version, but the "should evolve schema to second version" test
+  // should create a second version, which will change this to the subject name itself
+  let deletionConfirmation = "v1";
+
+  test.beforeEach(async ({ page, electronApp }) => {
+    subjectName = "";
+
+    // disable auto-formatting and language detection to avoid issues with the editor
+    // NOTE: this can't be done in a .beforeAll hook since it won't persist for each test run
+    await configureVSCodeSettings(page, electronApp, {
+      "workbench.editor.languageDetection": false,
+      "editor.autoClosingBrackets": "never",
+      "editor.autoClosingQuotes": "never",
+      "editor.autoIndent": "none",
+      "editor.autoSurround": "never",
+      "editor.formatOnType": false,
+      "editor.insertSpaces": false,
+      "json.format.enable": false,
+      "json.validate.enable": false,
+    });
```

Comment:
May need more adjustments here for the JSON tests that are having issues with `page.keyboard.type`:

![Image](https://github.com/user-attachments/assets/b5f7beae-8dd1-4fe8-9156-9afea3329195)

- https://playwright.dev/docs/api/class-locator#locator-fill may need a nested element to work properly
- https://playwright.dev/docs/api/class-locator#locator-press-sequentially shows the same behavior as `page.keyboard.type` above

Reply:
Adjusting this setting appears to have fixed the odd newline/comma behavior:
```
"editor.acceptSuggestionOnEnter": "off",
```